totes

```diff
@@ -1,5 +1,8 @@
 #! /usr/bin/env bash
 
+# ref - https://stackoverflow.com/a/3601734
+[[ -n ${1+x} ]] && optional_subject="$1"
```

Comment:
I would think we might prefer `${1:+x}` here.  If on the odd/off chance that $1 is set but null, we'd want the result to  be null (and so, not set optional_subject), rather than x.  Right?

Reference: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02

Not a particular fan of the magic global scope, either, but maybe this is just the Shell Script Way.

Reply:
Hmm so on second though I think I was overcomplicating this slightly and that `[[ -n "$1" ]]` will do what we want. Appreciate the attention to detail  👀 

I agree that shell scripts frequently have more global scope than I'd like sometimes. Are you suggesting you'd rather pass this in to the `get_notes_path` directly? I like renaming the arguments since I think it's confusing to see `$1` and try to remember what the first argument is. We could pass it in directly:

```bash
note_to_render="$(get_note_to_render "$optional_subject")" || exit $?
```

Inside the `get_note_to_render` `optional_subject` will still be defined though since I'm not aware of a way to restrict assignments in the calling scope. All in all I think it is the Shell Script Way to do this sort of thing though I'm not an expert so could definitely be missing something.

Reply:
yeah, I was thinking about passing it in, rather than assuming it was in the ether.  But maybe ether-access is the norm for shell scripts, so I don't have a strong opinion.

Fine with me to leave it as is.